blog/content/post/2020-04-29-trenchboot-uefi-environment.md

```diff
@@ -0,0 +1,833 @@
+    kernel during platform booting. It extends both SHA256 and SHA1 values.
+    However, the expected result is valid only for SHA256 if used with TPM2.0
+    device.
+
+    To properly execute script, first find correct directory to `bzImage` and
+    `initrd`. Best way to find exact directories is to see `"NixOS - Secure
+    Launch"` entry in `/boot/grub/grub.cfg`:
+
+    ```
+    $ cat /boot/grub/grub.cfg
+    (...)
+    menuentry "NixOS - Secure Launch" {
+      search --set=drive1 --fs-uuid 4881-6D27
+      slaunch skinit
+      slaunch_module ($drive1)//lz_header
+      linux ($drive1)//kernels/3w98shnz1a6nxpqn2wwn728mr12dy3kz-linux-5.5.3-bzImage systemConfig=/nix/store/ci38is4cvjlz528jay66h7qpqr6ws22n-nixos-system-nixos-20.09.git.c156a866dd7M init=/nix/store/ci38is4cvjlz528jay66h7qpqr6ws22n-nixos-system-nixos-20.09.git.c156a866dd7M/init console=ttyS0,115200 earlyprintk=serial,ttyS0,115200 console=tty0 loglevel=4
+      initrd ($drive1)//kernels/k1x969q4mwj59hyq3hn2mcxck8s2410a-initrd-linux-5.5.3-initrd
+    }
+    (...)
+    ```
+
+    `/nix/store/3w98shnz1a6nxpqn2wwn728mr12dy3kz-linux-5.5.3/bzImage`
+    is directory to Linux kernel.
+    `/nix/store/k1x969q4mwj59hyq3hn2mcxck8s2410a-initrd-linux-5.5.3/initrd` is
+    directory to initrd.
+
+4. Go to `/nix/store/` and run below command:
+
+    ```
+    $ cd /nix/store
+    $ ls | grep landing-zone
+    5a6kapnjxs8dj4jp49qagz1mw2r6hnr2-landing-zone-debug-0.3.0
+    l1b2h84fdw8g0m9aygmv8g3nhbnw9kic-landing-zone-debug-0.3.0.drv
+    lf763br9hm0ipp76k2p16iq75x3xpgrm-landing-zone-0.3.0
+    mnbh5xahlbzmfa50r60y5z4lph9rd41k-landing-zone-0.3.0.drv
+    ```
+
+    > Hash before `-landing-zone-0.3.0` is dependent on built version and might
+    > be different in yours. Choose non-debug version from above results.
+
+5. Go to `/nix/store/lf763br9hm0ipp76k2p16iq75x3xpgrm-landing-zone-0.3.0`
+   directory.
+
+    ```
+    $ cd /nix/store/lf763br9hm0ipp76k2p16iq75x3xpgrm-landing-zone-0.3.0
+    ```
+
+6. Execute `./extend_all.sh` script.
+
+    Usage is `./extend_all.sh <directory-to-bzImage> <directory-to-initrd>`
+
+    It must be executed inside the directory containing currently used (debug or
+    non-debug) version of `lz_header.bin`. You should already be in this
+    directory after the previous step. Directories to `bzImage` and `initrd` we
+    found in step 3.
+
+    ```
+    ./extend_all.sh /nix/store/3w98shnz1a6nxpqn2wwn728mr12dy3kz-linux-5.5.3/bzImage /nix/store/k1x969q4mwj59hyq3hn2mcxck8s2410a-initrd-linux-5.5.3/initrd
+    ff4562570be624792d7cc1ae8dcb8a2c9d978cfd  SHA1
+    21d3d024420a4149a3a226d39331a3a69e434200ee2d1c56fb02f3b982dc97b2  SHA256
+    ```
+
+    Compare SHA256 value with PCR17 content checked previously with
+    `tpm2_pcrread` output. If DRTM is enabled and executes properly, they
+    should be the same. It proves that the LZ code utilizes the SHA256 algorithm during
+    measurements.
+
+## Summary
+
```

Comment:
@miczyg1 I feel that we are missing 2-3 sentences of summary here. If this project's stage was hard to achieve? Any problems encountered? Or something similar.

Reply:
@artur-rs @krystian-hebel  As we mentioned earlier, we are introducing new platforms according to project's development. Moreover, in this blog post we have shown how to enable DRTM in UEFI environment as well. It proves that solution, proposed by us, is not only dedicated to a narrow group of devices, but it can be a common tool for many more. Therefore, keep follow our work and project's progress and feel free to experiment with your own hardware too!